### PR TITLE
Add support for layers and AWS credentials file

### PR DESCRIPTION
**Pull request by S-Cardenas**
I added support for [layers](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-layerversion.html#sam-layerversion-contenturi) (only local layers) specified in the AWS SAM template.

I also added the ability to specify the location of your AWS configuration and credentials file `~/.aws/credentials` so that the folder is mapped via a layer to your lambda containers and can be accessed by something like [AWS SharedInFileCredentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SharedIniFileCredentials.html).

### CODE REVIEW

**Comment by S-Cardenas** [comment]
@duartemendes @cjol @Kivol Any update on this PR? Please and thanks 👍

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "aws-sam-api-proxy",
-  "version": "0.0.20",
+  "name": "@s-cardenas/aws-sam-api-proxy",
```
Comment (duartemendes) [inline package.json]: We need to undo this change both here and on the `package-lock..json`.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+  "version": "0.0.31",
```
Comment (duartemendes) [inline package.json]: Let's undo this change, the version will be updated separately from this PR along with an update on the changelog, what do you think?

File: tests/controller/index.spec.js [hunk 3]
```diff
@@ -23,13 +23,15 @@ describe('controller', () => {
...
       end: jest.fn().mockReturnThis(),
+      setHeader: jest.fn().mockReturnThis(),
     };
 
     httpClientStub = {
       post: jest.fn(),
     };
 
     controller = createController(httpClientStub, functionsWithReplicatedValue);
+    console.log('controller: ', controller);
```
Comment (duartemendes) [inline tests/controller/index.spec.js]: Was this left here by mistake?

File: src/serverlessFunctions/parseFromTemplate.js [hunk 4]
```diff
@@ -23,8 +23,32 @@ const buildFnPathData = (path) => {
   };
 };
 
+const buildLayersConfig = (basePath, templatePath, template, prelimLayers) => {
+  const config = [];
+
+  if (prelimLayers && prelimLayers.length > 0) {
+    prelimLayers.forEach((layer) => {
+      if (!('Ref' in layer)) throw new Error('Only referenced layers are supported.');
```
Comment (duartemendes) [inline src/serverlessFunctions/parseFromTemplate.js]: Won't this be a breaking change and a blocker for people using layers without `Ref`? 🤔 

I wonder if in these cases we should simply log a warning and skip the current layer being processed, what do you think?

```suggestion
      if (!('Ref' in layer)) {
        console.warn('Only referenced layers are supported, skipping layer', layer);
        return;
      }
```

Or, do you reckon this behavior could be misleading?

File: src/serverlessFunctions/parseFromTemplate.js [hunk 5]
```diff
@@ -23,8 +23,32 @@ const buildFnPathData = (path) => {
...
+
+      const layerName = layer.Ref;
+      const cftLayer = template.Resources[layerName];
+      const layerContentUri = cftLayer.Properties.ContentUri;
```
Comment (duartemendes) [inline src/serverlessFunctions/parseFromTemplate.js]: According to the [docs](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-layerversion.html#sam-layerversion-contenturi), `ContentUri` might be an object, which will cause the following line to break as we try to merge strings with an object.

Should we apply some validation?

**Comment by duartemendes** [comment]
> @duartemendes @cjol @Kivol Any update on this PR? Please and thanks 👍

@S-Cardenas so sorry for taking so long, unfortunately, I was not notified at the moment of the PR creation. Next time, please add me as a reviewer or tag me as you did now.

Thank you so much for your work on this, it's very appreciated.

I've left a few small comments/questions. I'm happy to merge the PR and create a new release once we go through those conversations.

Also, it would be very interesting to have the `README` file updated, briefly explaining the new feature!

File: src/serverlessFunctions/parseFromTemplate.js (hunk 4 above)
Comment (S-Cardenas) [reply src/serverlessFunctions/parseFromTemplate.js]: That's a good catch. I agree.

File: tests/controller/index.spec.js (hunk 3 above)
Comment (S-Cardenas) [reply tests/controller/index.spec.js]: This was a debugging mistake. I'll remove it.

File: src/serverlessFunctions/parseFromTemplate.js (hunk 5 above)
Comment (S-Cardenas) [reply src/serverlessFunctions/parseFromTemplate.js]: Also a very good catch. I will only only this layer to be processed if it's ContentUri is a relative path (i.e. string)

File: package.json (hunk 1 above)
Comment (S-Cardenas) [reply package.json]: Done.

File: package.json (hunk 2 above)
Comment (S-Cardenas) [reply package.json]: Agree and done.

**Comment by S-Cardenas** [comment]
@duartemendes Thank you for the review. I have implemented all of your suggestions mentioned above.

Additionally I have added support for the option `--docker-socket-path` in order to allow Linux users using Docker Desktop to specify their own custom docker socket path (see the [docker docs](https://docs.docker.com/desktop/faqs/linuxfaqs/#context) for why Linux users running the desktop app have a different endpoint).

**Review by duartemendes — APPROVED** [review]
Thanks again for the contribution @S-Cardenas 🚀
